surface/src/simplification_remove_unused_vertices.cpp

```diff
@@ -55,12 +55,12 @@ pcl::surface::SimplificationRemoveUnusedVertices::simplify(const pcl::PolygonMes
   indices.reserve (nr_points);
 
   // mark all points in triangles as being used
-  for (size_t polygon = 0; polygon < input.polygons.size (); ++polygon)
-    for (size_t point = 0; point < input.polygons[polygon].vertices.size (); ++point)
-      if (new_indices[ input.polygons[polygon].vertices[point] ] == -1 )
+  for (const auto &polygon : input.polygons)
+    for (size_t point = 0; point < polygon.vertices.size (); ++point)
```

Comment:
This feels like it can also be converted into a for-range style loop. There's no isolated use of the index `point`.